### ISSUE TITLE
Verify unique of array got panic when the array item is type of object

### ISSUE THREAD

**Issue by emnnqqs** [getkin/kin-openapi#143]
I have defined below array in openapi:
```
    Tests:
      type: array
      items:
        $ref: '#/components/schemas/Test'
      minItems: 0
      uniqueItems: true
    Test:
      type: object
      properties:
        attri1:
          type: integer
        attri2:
          type: integer
```
When i verify it by openapi3filter, I got below error:
```
panic: runtime error: hash of unhashable type map[string]interface {} [recovered]
	panic: runtime error: hash of unhashable type map[string]interface {}

goroutine 8 [running]:
testing.tRunner.func1(0xc0000d8200)
	C:/Go/src/testing/testing.go:874 +0x3aa
panic(0x7ea040, 0xc0001a6970)
	C:/Go/src/runtime/panic.go:679 +0x1c0
github.com/getkin/kin-openapi/openapi3.isSliceOfUniqueItems(0xc0001af380, 0x2, 0x2, 0x1)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:1227 +0x187
github.com/getkin/kin-openapi/openapi3.(*Schema).visitJSONArray(0xc00019a000, 0xc0001af380, 0x2, 0x2, 0x0, 0x0, 0x2dd0008)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:985 +0x8f2
github.com/getkin/kin-openapi/openapi3.(*Schema).visitJSON(0xc00019a000, 0x7bbca0, 0xc0001af3a0, 0xc00008d600, 0xb7d770, 0xc00008d5f0)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:624 +0x5e2
github.com/getkin/kin-openapi/openapi3.(*Schema).VisitJSON(0xc00019a000, 0x7bbca0, 0xc0001af3a0, 0xa, 0xc00008d6b0)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:593 +0x4b
github.com/getkin/kin-openapi/openapi3.(*Schema).visitJSONObject(0xc0000cc8c0, 0xc0001a8d20, 0xc0001a8d00, 0xc0001a6800, 0x0)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:1082 +0x79a
github.com/getkin/kin-openapi/openapi3.(*Schema).visitJSON(0xc0000cc8c0, 0x7df880, 0xc0001a8d20, 0x812a00, 0x1, 0xc0001a8cf0)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:626 +0x4ae
github.com/getkin/kin-openapi/openapi3.(*Schema).VisitJSON(0xc0000cc8c0, 0x7df880, 0xc0001a8d20, 0xc0001ae0e0, 0xc0001a6850)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3/schema.go:593 +0x4b
github.com/getkin/kin-openapi/openapi3filter.ValidateRequestBody(0x8ee6c0, 0xc000012158, 0xc0001a8c00, 0xc0001a8090, 0x0, 0x0)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3filter/validate_request.go:191 +0x45c
github.com/getkin/kin-openapi/openapi3filter.ValidateRequest(0x8ee6c0, 0xc000012158, 0xc0001a8c00, 0xc00008b580, 0xc0000a7270)
	C:/Users/emnnqqs/git-repository/gopath/src/github.com/getkin/kin-openapi/openapi3filter/validate_request.go:58 +0x3ba
```

**Comment by fenollp** [comment]
Indeed since the code that checks for unique items is this: https://github.com/getkin/kin-openapi/blob/1116f3ecaa880095aa0e8f4eb64a6dc04c8a288e/openapi3/schema.go#L1222 which requires that the things being checked must be comparable (must be map keys) this means any slice or map (object) cannot be given to that implementation.

A quick workaround can be to edit this function to encode keys into strings, assuming JSON encoding of values always has the same string representation given the same input. I'll gladly review such a PR along with tests ;)

**Comment by Arms-Li** [comment]
> Indeed since the code that checks for unique items is this:
> 
> https://github.com/getkin/kin-openapi/blob/1116f3ecaa880095aa0e8f4eb64a6dc04c8a288e/openapi3/schema.go#L1222
> 
> which requires that the things being checked must be comparable (must be map keys) this means any slice or map (object) cannot be given to that implementation.
> A quick workaround can be to edit this function to encode keys into strings, assuming JSON encoding of values always has the same string representation given the same input. I'll gladly review such a PR along with tests ;)

I have a quick think that we can use fmt.Sprintf("%v", key) to convert all keys into strings.  And the codes look like below:
```
func isSliceOfUniqueItems(xs []interface{}) bool {
	s := len(xs)
	m := make(map[string]struct{}, s)
	for _, x := range xs {
		m[fmt.Sprintf("%v", x)] = struct{}{}
	}
	return s == len(m)
}
```
I will have local test first, if everything is ok, I will create a PR.

**Comment by fenollp** [comment]
Fixed by #145
